### PR TITLE
Change the deprecated `collections`

### PR DESCRIPTION
**Pull request by belldandyxtq**
This commit solves the Issue #6637. It removes the deprecated usage
of abc from `collections`, which will be removed in Python3.8.

The abc is now imported as `from chainer.utils import collections_abc`,
which aligns with `chainer` code base.

For more details of deprecated usage of abc in collections, please refer
to issue #6637.

Thank you for creating a pull request!

Please double-check the following.

- Read [our contribution guide](https://docs.chainer.org/en/stable/contribution.html).
  - Does your code conform to our coding guidelines?
  - Did you write sufficient test code?
  - Did you write sufficient documentation?
- Also, take a look at [our compatibility policy](https://docs.chainer.org/en/stable/compatibility.html).

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins, test this please.

File: chainermn/communicators/mpi_communicator_base.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import collections
+from chainer.utils import collections_abc
```
Comment (toslunar) [inline chainermn/communicators/mpi_communicator_base.py]: Could you move the line to the next to the line `import chainer.utils`?

Comment (toslunar) [inline chainermn/communicators/mpi_communicator_base.py]: Could you move the line to the next to the line `import chainer.utils`?

Comment (belldandyxtq) [reply chainermn/communicators/mpi_communicator_base.py]: Done, thanks you for the suggestion

**Comment by belldandyxtq** [comment]
Jenkins, test this please.

**Pull request merged**